### PR TITLE
BookieWatcherImpl doesn't initCause when throwing MetaStoreException

### PR DESCRIPTION
**Pull request by codingwangqi**
Descriptions of the changes in this PR:

### Motivation

when BookieWatcherImpl throws MetaStoreException, it doesn't provide the cause. so when the exception is logged, we can't tell the actual failure cause.

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [ ] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java [hunk 1]
```diff
@@ -62,7 +62,9 @@
             log.error("Interrupted reading bookie list : ", cause);
             return new BKInterruptedException();
         } else {
-            return new MetaStoreException();
+            MetaStoreException mse = new MetaStoreException();
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java]: We should add a constructor which takes directly the cause

Comment (codingwangqi) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java]: @eolivelli it seems that I have to change a lot for that. I am new to the project not sure if I should do that. what is the concern regarding using initCause?

Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java]: MetaStoreException already has a constructor which takes the 'cause'.

It is cleaner to use that one and not call initCause

Comment (codingwangqi) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java]: @eolivelli I don't think so. MetaStoreException is BKException, BKException doesn't have a constructor with Throwable. but I will try to change it as what you suggested.

Comment (eolivelli) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookieWatcherImpl.java]: Oh, don't worry.
I don't have my laptop no
Yesterday I tried to look for the constructor using github web ui and maybe I was wrong and I looked to the wrong class.

**Comment by sijie** [comment]
run integration tests

**Comment by sijie** [comment]
run integration tests

**Pull request closed without merging**